vita3k/modules/SceJpeg/SceJpegUser.cpp

```diff
@@ -112,14 +152,21 @@ EXPORT(int, sceJpegGetOutputInfo, const uint8_t *jpeg_data, uint32_t jpeg_size,
     TRACY_FUNC(sceJpegGetOutputInfo, jpeg_data, jpeg_size, format, mode, output);
     const auto state = emuenv.kernel.obj_store.get<MJpegState>();
 
+    if (!jpeg_data || !output) {
+        return SCE_JPEG_ERROR_INVALID_POINTER;
+    }
+    if (!jpeg_size) {
+        return SCE_JPEG_ERROR_OUT_OF_MEMORY;
+    }
+
     DecoderSize size = {};
 
     state->decoder->send(jpeg_data, jpeg_size);
     state->decoder->receive(nullptr, &size);
 
     output->width = size.width;
     output->height = size.height;
-    output->color_space = 2u << 16u;
+    output->color_space = convert_color_space_decoder_to_jpeg(state->decoder->get_color_space());
     output->output_size = size.width * size.height * 3;
```

Comment:
It looks like this value is also wrong. Format looks like it can only be equal to 0 or -1 (otherwise it will return an error). If format is 0, then the output is the rgba size (so size.width * size.height * 4), otherwise (format is -1) it is the YUV size, which is less.

Reply:
I've updated the code to reflect your advice. There appears to be additional behavior based on mode, but I haven't identified it correctly yet.